kart/raster/v1.py

```diff
@@ -163,3 +195,135 @@ def diff_to_working_copy(
         ds_diff["tile"] = tile_diff
 
         return ds_diff
+
+    def _non_pam_wc_paths(self, tile_and_pam_paths):
+        """
+        Given a list of dirty paths, relative to the workdir -
+        return the paths of the tiles (relative to the workdir) that have been affected
+        either by editing directly or by editing their PAM files.
+        """
+        result = set()
+        for path in tile_and_pam_paths:
+            if path.lower().endswith(".aux.xml"):
+                tile_paths = find_similar_files_case_insensitive(
+                    self._workdir_path(path[:-8])
+                )
+                wc_path = self._workdir_path("")
+                result.update(str(t.relative_to(wc_path)) for t in tile_paths)
+            else:
+                result.add(path)
+        return result
+
+    @property
+    def tile_metadata(self):
+        return {
+            "format.json": self.get_meta_item("format.json"),
+            "schema.json": self.get_meta_item("schema.json"),
+            "crs.wkt": self.get_meta_item("crs.wkt"),
+            **self.get_meta_items_matching(self.BAND_RATS),
+            **self.get_meta_items_matching(self.BAND_CATEGORIES),
+        }
+
+    @property
+    def tile_count(self):
+        """The total number of tiles in this dataset, not including PAM files."""
+        if self.inner_tree is not None:
+            try:
+                subtree = self.inner_tree / self.TILE_PATH
+                return sum(
+                    1
+                    for blob in all_blobs_in_tree(subtree)
+                    if not blob.name.endswith(".aux.xml")
+                )
+            except KeyError:
```

Comment:
what's this catching exactly?

Reply:
No such subtree at `inner_tree / "tile"`
I've rearranged it so that the KeyError is only caught where its useful, instead of over the whole function